tests/wallet/sync/test_wallet_sync.py

```diff
@@ -27,6 +31,67 @@ def wallet_height_at_least(wallet_node, h):
 
 
 class TestWalletSync:
+    @pytest.mark.asyncio
+    async def test_request_block_headers(self, bt, wallet_node, default_1000_blocks):
+        # Tests the edge case of receiving funds right before the recent blocks  in weight proof
+        full_node_api: FullNodeAPI
+        full_node_api, wallet_node, full_node_server, wallet_server = wallet_node
+
+        wallet = wallet_node.wallet_state_manager.main_wallet
+        ph = await wallet.get_new_puzzlehash()
+        for block in default_1000_blocks[:100]:
+            await full_node_api.full_node.respond_block(full_node_protocol.RespondBlock(block))
+
+        msg = await full_node_api.request_block_headers(wallet_protocol.RequestBlockHeaders(10, 15, False))
+        assert msg.type == ProtocolMessageTypes.respond_block_headers.value
+        res_block_headers = RespondBlockHeaders.from_bytes(msg.data)
+        bh = res_block_headers.header_blocks
+        assert len(bh) == 6
+        assert [x.reward_chain_block.height for x in default_1000_blocks[10:16]] == [
+            x.reward_chain_block.height for x in bh
+        ]
+
+        assert [x.foliage for x in default_1000_blocks[10:16]] == [x.foliage for x in bh]
+
+        assert [x.transactions_filter for x in bh] == [b"\x00"] * 6
+
+        num_blocks = 20
+        new_blocks = bt.get_consecutive_blocks(
+            num_blocks, block_list_input=default_1000_blocks, pool_reward_puzzle_hash=ph
+        )
+        for i in range(0, len(new_blocks)):
+            await full_node_api.full_node.respond_block(full_node_protocol.RespondBlock(new_blocks[i]))
+
+        msg = await full_node_api.request_block_headers(wallet_protocol.RequestBlockHeaders(110, 115, True))
+        res_block_headers = RespondBlockHeaders.from_bytes(msg.data)
+        bh = res_block_headers.header_blocks
+        assert len(bh) == 6
+
+    @pytest.mark.asyncio
+    async def test_request_block_headers_rejected(self, bt, wallet_node, default_1000_blocks):
```

Comment:
you can use `pytest` parameterization to boil down this test to its essence, and have it run over all parameters

Reply:
I think this would still make sense. you can parameterize it on tuples of (start_height, end_height, expected_response). At least for the tests of `request_block_headers()`

Reply:
I tried it, it works, but it makes the test slower here, since we are adding 150 blocks to the chain. It might not be worth the performance hit